include/boost/geometry/views/closeable_view.hpp

```diff
@@ -44,19 +47,23 @@ struct closing_view
 {
     // Keep this explicit, important for nested views/ranges
     explicit inline closing_view(Range& r)
-        : m_range(r)
+        : m_range(boost::ref(r))
     {}
 
     typedef closing_iterator<Range> iterator;
     typedef closing_iterator<Range const> const_iterator;
 
-    inline const_iterator begin() const { return const_iterator(m_range); }
-    inline const_iterator end() const { return const_iterator(m_range, true); }
+    inline const_iterator begin() const { return const_iterator(cref()); }
+    inline const_iterator end() const { return const_iterator(cref(), true); }
+
+    inline iterator begin() { return iterator(ref()); }
+    inline iterator end() { return iterator(ref(), true); }
 
-    inline iterator begin() { return iterator(m_range); }
-    inline iterator end() { return iterator(m_range, true); }
 private :
-    Range& m_range;
+    inline Range& ref() { return m_range; }
+    inline Range const& cref() const { return m_range; }
+
+    boost::reference_wrapper<Range> m_range;
 };
 
```

Comment:
Changes in this file (and probably identity_view) are not related to the problem at hand.

Hope you understand, but these reviews are quite large and it is confusing to see various unrelated changes here. Or even if they are related, but necessary first steps, I would prefer to see them separately as one consistent PR with accompagnying unit tests.

Anyway, my question is: what do the changes this exactly improve? I don't see at a glance in the unit tests a geometry requiring a reference_wrapper instead of a wrapper.